lib/core/refresh.js

```diff
@@ -0,0 +1,246 @@
+import jwtDecode, { InvalidTokenError } from 'jwt-decode'
+import { getProp } from './utilities'
+import ExpiredAuthSessionError from './includes/ExpiredAuthSessionError'
+
+export class RefreshController {
+  constructor (scheme) {
+    this.scheme = scheme
+    this.refreshPromise = null
+    this.refreshInterval = undefined
+  }
+
+  // Returns a promise which is resolved when refresh is completed
+  // Call this function when you intercept a request with an expired token.
+  // Multiple requests will be queued until the first has completed token refresh.
+  handleRefresh () {
+    // Another request has started refreshing the token, wait for it to complete
+    if (this.refreshPromise) {
+      return this.refreshPromise
+    }
+
+    return this._doRefresh()
+  }
+
+  _doRefresh () {
+    this.refreshPromise = new Promise(resolve => {
+      this.scheme._refreshToken().then(response => {
+        this.refreshPromise = null
+        resolve(response)
+      })
+    })
+
+    return this.refreshPromise
+  }
+
+  initializeAutomaticRefresh () {
+    // If auto refresh is enabled, schedule token refresh
+    if (this.scheme.options.autoRefresh.enable) {
```

Comment:
Why not simply supporting `options.autoRefresh: false`?

Reply:
What do you think about add this to scheme construct to support mixed: 
```
if (typeof this.options.autoRefresh !== 'object') {
    this.options.autoRefresh.enable = !!this.options.autoRefresh`
}
```

Reply:
In case `autoRefresh` is disabled, we don't need other options (and defaults) so can be removed from template :) Also why introduce more complexity?

Reply:
Ok

Reply:
Fixed in commit https://github.com/nuxt-community/auth-module/pull/548/commits/95c61ce4ead21a9c011afed2bf4e2046ef3be37d